runtime/doc/options.txt

```diff
@@ -4390,6 +4390,28 @@ A jump table for the options with a short description can be found at |Q_op|.
 	    |nobody         |  3 nobody     |  0 nobody     |3   nobody
 	    |there          |  4 there      |  1 there      |  1 there
 
+				*'numbercolumn'* *'nuc'*
+'numbercolumn' 'nuc'	string	(default: empty)
+			local to window
+	EXPERIMENTAL
+	When this option is not empty, it determines the content of the number
+	column. The format of this option is like that of 'statusline'. The
+	variables *v:redraw_lnum* and *v:redraw_relnum* will be set to the to be printed line
```

Comment:
We can re-use `%l` instead of introducing a new variable. For relative line number we can either use a free letter (no obvious choices) or just overload and existing one (maybe `%r`) since most of them aren't applicable here.

Reply:
Consider that `%l` would write the cursor line number for each line. That would have to be overloaded too for it to return the actual number line to be drawn. But yeah overloading seems reasonable since most of seem irrelevant for the number column.

Reply:
Hmm, just realized `%l` can't be used within `%{ }` so a `v:` variable will be necessary.

Reply:
Yes, that is an unfortunate limitation but probably hard to overcome. Unless if we somehow pre-parse the %{} block for restricted characters to substitute the line number. Not sure that's feasible or worth it.

Reply:
Again, I think there could still be merit in adding/overloading stl items for 'cursorcolumn' for simple cases like `'set nuf=%=%l'`. But yeah I think the variables need to stay regardless.